st2client/st2client/commands/action.py

```diff
@@ -1489,13 +1492,27 @@ def tail_execution(cls, execution_manager, stream_manager, execution, output_typ
             print('Execution %s has completed (status=%s).' % (execution_id, execution.status))
             return
 
-        events = ['st2.execution__update', 'st2.execution.output__create']
+        # We keep track of all the workflow executions which could contain children
```

Comment:
Not a huge fan of all the code inside this function, but at least we now have test cases for all the various scenarios and edge cases.

Hopefully in the future we can come up with a more simplified code (perhaps by moving some of that filtering to the server as mentioned inside a comment).